core/src/bytes/BytesReader.h

```diff
@@ -156,6 +189,23 @@ namespace ledger {
 
 
         private:
+
+            /**
+             * Read a value (int, long, short, byte, struct) from the reader using big endian bytes ordering.
+             * @return
+             */
+            template<typename T> T readNextBeValue() {
+                T result;
+                uint8_t* ptr = reinterpret_cast<uint8_t *>(&result);
+                for (auto i = 0; i < sizeof(result); i++) {
+                    ptr[i] = readNextByte();
+                }
+                ledger::core::endianness::swapToEndianness(ptr, sizeof(result),
+                                                            ledger::core::endianness::Endianness::BIG,
+                                                            ledger::core::endianness::getSystemEndianness());
+                return result;
+            }
```

Comment:
I think you can generalise a little more that generic member 

```suggestion
            template<typename T, ledger::core::endianness:Endianness = ledger::core::endianness::Endianness::BIG> T readNextBeValue() {
                T result;
                uint8_t* ptr = reinterpret_cast<uint8_t *>(&result);
                for (auto i = 0; i < sizeof(result); i++) {
                    ptr[i] = readNextByte();
                }
                ledger::core::endianness::swapToEndianness(ptr, sizeof(result),
                                                            Endianness,
                                                            ledger::core::endianness::getSystemEndianness());
                return result;
            }
```